Spd window

flinkeKoralle opened the pull request:
<!-- Give a brief summary of your changes in the title. -->
Speed Window Preselection Function

### Description of Changes
<!-- Describe your changes in detail. -->
When in managed speed, rotating the speed selector knob will open the window for a certain time.
If this speed is pulled, than speed becomes selected, otherwise it returns to managed.
Also there is a difference between an open window with active managed speed (yellow dot) and just a closed window (dahes).
Furthermore selected speed is inhibited on ground, once engines are running.

### Screenshots (optional)

### Bugs fixed (if any)
<!-- If you fixed any bugs, describe them here. State issue number if applicable. -->
FCU speed window
FCU altitude selection of 100/1000

### Checklist:
<!-- [ ] = Unchecked, [x] = Checked. -->
* [x ] My changes follow the Contributing Guidelines. <!-- See CONTRIBUTING.md to verify. -->
* [x ] My changes implement realistic features. <!-- Only aircraft changes require this. -->
* [ ] Please have a main Developer test my changes before merging. <!-- We will always briefly test, but if it needs a "full" test, please check). -->

<!-- If you changes are not ready for merging, please submit this as a draft pull request. If it is ready, submit it as a regular pull request. -->
ready for merge

Commit:
retrieve dep.elevation and print

Commit:
before dashes

Commit:
accelAltFt

Commit:
created setAccelAlt func

Commit:
clr key on init and perf/to working

Commit:
conform tour 6.10.23

Commit:
split line for TOPERF page for thrred/accel

Commit:
Accel/ThrRed working as tested

Commit:
Merge branch 'dev' into mcdu-fmgc-merger

Commit:
corrected manual thr/acc alt with no depApt

Commit:
introducing ga-accel and ga-thrRed variables

Commit:
print accel/thrred accordingly

Commit:
ga accel/trhRed vars working- MCDU not correct

Commit:
trying to change fonrt size in perfga

Commit:
cannot change fontsize prfga page

Commit:
input in perfga almost working

Commit:
check if acc > thrRed

Commit:
round error corrected in acc/thrRed

Commit:
Startpoint for GA Property insertion

Commit:
improved imput on perf-ga page

Commit:
setting fonts

Commit:
input on perf ga further improved

Commit:
fadec thrRed changed in ga by variables

Commit:
minor errors corrected

Commit:
adding comments

Commit:
Minor change to test push request

Commit:
testing a timer with FD1 button

Commit:
default fcu alt set to 100 ft

Commit:
fcu alt selection for 1k position corrected

Commit:
finding FCU outputs

Commit:
Merge https://github.com/legoboyvdlp/A320-family into PERF-GA-Page-Accel/ThrRed-Input-correction

Commit:
Merge https://github.com/legoboyvdlp/A320-family into FCU-Preselect

Commit:
spd preselection timer in managed
todo: separate fcu selected speed and autoflight commanded speed

Commit:
preselection working
todo: spd adjust in managed gets fmgc speed instead of current

Commit:
something wrong here

Commit:
spd window indication corrected
todo: fcu default spd is 0. should be 100.

Commit:
fcu speed window corrected
todo: if FD1+2 off then revert to selected speed

Commit:
working commit. adding spd window functionality
todo: implement close/open spd window in FCU.nas file, simplifying the a320.flightdeck.xml file

Commit:
startpoint for new managed speed selection
todo: remove custom.input.spdManaged -> not needed
managed speed stops by itself if not valid

Commit:
not working to get back to selected

Commit:
interim: fms gets into selected when spd pushed
todo: check when fcu-spd is reset to selected

Commit:
fmgc logic and fcu working todo: fcu cleanup and spd window

Commit:
separation of mngd spd active and valid
todo: display correctly on fcu
show dashes and no dot when in done phase

Commit:
working on fcu indication at startup
todo: phase transition check (done phase)
and redundant variables

Commit:
start to refine FCU spd window

Commit:
selected spd on gnd inhibited

Commit:
mngSpd triggered by FDs on gnd

Commit:
mng speed triggered by v2 on gnd
TODO: mng spd triggered by to/mct

Commit:
before SRS trigger

Commit:
Managed speed on gnd as expected.
TODO: selected spd only after 5s airborne

Commit:
sel spd inhibit until 5s after lift-off
TODO: sel spd disarms on gnd at eng start
sel spd not poss. on gnd with engs running and fd on.

Commit:
Merge branch 'FCU-Preselect' into SPD-Window

Commit:
Merge branch 'PERF-GA-Page-Accel/ThrRed-Input-correction' into SPD-Window

Commit:
sel spd inhibit with one eng running

Commit:
align on gnd
TODO: spd push on gnd, if pos lost no mangd spd

Commit:
FCU SPD window behaviour improved
TODO: move Mach/Speed changeover to a property and activate via listener.

Octal450 commented:
That's a lot of changes, will need thorough testing.

I can't promise a timeline on that right now but I'll add it to my list.

Kind Regards,
Josh

flinkeKoralle commented:
Hi Josh,

You can ask me about all the changes - I guess it is faster in some
cases.

One change that is very easy to see is the one in FCU.nas, the
ALTAdjust function.
It changes the behavior for the knob to the real one (round to nearest
1000 when selector set to 1000).

Another change was the separation of the thrust reduction at take off
and the one for the G/A.
(Set on PERFGA.nas file and preset as an Pin Option on the DLH.xml).

The biggest one was the behavior of the speed window.
I think there is still room for improvement, as the managed speed is
embedded in a loop.
I hope I can change the managed speed to fully event based
(setlisteners). But it is ok for now.
Latest, when more complex failures kick in, it would be easier to
implement the real behavior if conditions are not nested in a loop I
think.

I chose to improve the speed window behavior because I thought it would
be easy.
It turned out to be much more complex. But I learned a lot.

I saw the post with the autobrake problem.
I thought about implementing it, but I think it would be a lot of work,
because I would implement the full LGCIU modules and its signals.

At the moment I want to improve a little more the managed/selected
speed. And from there maybe a failure case.

Greetings back,
Max

On Mon, 2024-02-05 at 09:02 -0800, Josh Davidson wrote:
> That's a lot of changes, will need thorough testing.
> I can't promise a timeline on that right now but I'll add it to my
> list.
> Kind Regards,
> Josh
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>

Commit:
Cleaned up Managed Speed vars. Preparing to merge last redundant vars:
fmgc.Custom.Input.spdManaged has same functionality as fmgc.FMGCNodes.mngSpdActive

Commit:
consolidated all managed speed into mngSpdActive, and spdWindowOpen

Commit:
removed unused "fmgcSpeedValid"
